python/cernbox_utils/sharing.py

```diff
@@ -5,11 +5,19 @@
 from .script import get_eos_server_string, get_eos_server
 
 # remove duplicates, preserving order
-def squash(seq):
-    seen = set()
-    seen_add = seen.add
-    return [x for x in seq if not (x in seen or seen_add(x))]
+# use the most permissive acl (otherwise EOS picks the first one)
+def squashAcls(acls):
```

Comment:
I changed the squash behavior. Before we deleted repeated ACls, so the same user could have 2 different ACLs i.e u123:rx,u123:rwx+d). And, on EOS, if you set both, it only keeps one of those.

Now I keep only the most permissive ACLs for a user (u123:rx,u123:rwx+d -> u123:rwx+d).

Reply:
I detected this because in a testing account I had:

/SHARES -> shared RW with A (u123:rwx+d)
/SHARES/SHARE1 -> shared RO with A  (u123:rx)

And the fix was having problems setting the proper acls. The expected whould be RW permissions in both, since SHARE1 is inside SHARES.

Reply:
Good, I think EOS should also apply the most permissive ACLs. I'm not sure 100% about it.

Reply:
The fix was failing because it was applying the rx instead of the rwx. I think (but haven't tested) that EOS was taking them in order and applying the first in the list.

Reply:
@diocas is not clear to me, if having rx and rwx which one will the script apply?